File: static/code/sample/tethered/gdl-setemphmd-b.cs
// In this sample:
// Read temperature and humidity every 1.5 second

using System.Data.SqlTypes;
using GHIElectronics.DUELink;

var availablePort = DUELinkController.GetConnectionPort();
var duelink = new DUELinkController(availablePort);

float GetTemperature() {
    return duelink.Engine.ExecuteCommand("Temp(SenPin(), SenType())");
}

float GetHumidity() {
    return duelink.Engine.ExecuteCommand("Humid(SenPin(), SenType())");
}

while (true) {   
    Console.WriteLine($"Temperature: {GetTemperature().ToString()} ");
    Console.WriteLine($"Humidity: {GetHumidity().ToString()} ");

    Thread.Sleep(1500);
}

File: static/code/sample/tethered/gdl-setemphmd-b.py
# In this sample:
# Read temperature and humidity every 1.5 second

import time
from datetime import datetime
from DUELink.DUELinkController import DUELinkController

availablePort = DUELinkController.GetConnectionPort()
duelink = DUELinkController(availablePort)

# method
def GetTemperature():
    return float(duelink.Engine.ExecuteCommand("Temp(SenPin(), SenType())"))

def GetHumidity():
    return float(duelink.Engine.ExecuteCommand("Humid(SenPin(), SenType())"))

while True:
    print(f"Temperature: {GetTemperature()} ")
    print(f"Humidity: {GetHumidity()} ")

    time.sleep(1.5)  # 1500 ms
